database/migrations/platform/2024_05_21_000000_update_siteawards_table.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::table('SiteAwards', function (Blueprint $table) {
+            $table->dropIndex(['User']);
+        });
+        Schema::table('SiteAwards', function (Blueprint $table) {
+            $table->dropUnique(['User', 'AwardData', 'AwardType', 'AwardDataExtra']);
+        });
+        Schema::table('SiteAwards', function (Blueprint $table) {
+            $table->dropColumn('User');
+        });
+        Schema::table('SiteAwards', function (Blueprint $table) {
+            $table->index(['user_id', 'AwardData', 'AwardType', 'AwardDataExtra']);
+        });
+    }
+
+    public function down(): void
+    {
+        Schema::table('SiteAwards', function (Blueprint $table) {
+            $table->dropIndex(['user_id', 'AwardData', 'AwardType', 'AwardDataExtra']);
+            $table->string('User', 50)->after('AwardDate');
+            $table->unique(['User', 'AwardData', 'AwardType', 'AwardDataExtra'], 'siteawards_user_awarddata_awardtype_awarddataextra_unique');
```

Comment:
This is failing for me. Presumably because User is null for every record.

Reply:
This probably can't be resolved in the migration itself and requires manual intervention.

```sql
UPDATE SiteAwards sa
JOIN UserAccounts ua ON sa.user_id = ua.ID
SET sa.User = ua.User;
```

Reply:
Agreed, but having an exception is bad. Recommend just not recreating the index.

Reply:
Got it - I wasn't sure if that was a viable approach. Updated in latest to not recreate the index.